### ISSUE TITLE
Input validation

### ISSUE THREAD

**Issue by carcigenicate** [floybix/cljbox2d#5]
I'm not sure if this is even maintained any more, but input validation is really needed here.

My entire experience with this library so far has been a string of NPEs coming about because I gave a vector when I should have given a number (or vice versa), or gave a bad key to  a map.

The failure's all happen in `step!` and give no indication of what *actually* is wrong, which is frustrating.

The port is nice, but it's proving to be difficult to use.

**Comment by floybix** [comment]
Fair enough... if I wrote it today I would spec the functions so you could validate inputs with spec/instrument.

To be honest I'm not likely to tackle this in the near future. If I do it would be in the newer `liquidfun` project.

Another interesting option for physics with clojure is [Arcadia](https://github.com/arcadia-unity/Arcadia).
